lib/SILOptimizer/LoopTransforms/LICM.cpp

```diff
@@ -504,7 +511,7 @@ static void analyzeBeginAccess(BeginAccessInst *&BI,
   };
 
   if (std::all_of(BeginAccesses.begin(), BeginAccesses.end(), safeBeginPred)) {
-    HoistingPairs.push_back(std::make_pair(BI, EI));
+    HoistingPairs.push_back(std::make_pair(BI, matchingEnds));
```

Comment:
There's no reason to copy the set here. I would use `emplace_back` to avoid creating a temporary pair, and std::move to avoid copying `matchingEnds`. I'm not sure if that's the best way to do it, but it makes sense to me.